src/Components/NotFoundErrorPage.tsx

```diff
@@ -43,3 +60,27 @@ export const NotFoundErrorPage = connect(function NotFoundErrorPage() {
     </>
   )
 })
+
+const NoRootError = connect(
+  function NoRootError() {
+    Obj.root() // read root again, to trigger "loading" state
+
+    useEffect(() => {
+      load(() => Obj.root()).then((root) => {
+        if (root === null && !isUserLoggedIn()) ensureUserIsLoggedIn()
+      })
+    }, [])
+
+    return <div>Page not found.</div>
+  },
+  { loading: () => null },
+)
+
+const PortalSiteNotFound = connect(
```

Comment:
* [ ] FYI to be more inline with the naming of the other components in this module: `NoSite`.